_src/pattern-library/sass/components/_forms.scss

```diff
@@ -0,0 +1,484 @@
+    &.has-success {
+        border-color: palette(success, base);
+
+        & + .field-hint {
+            color: palette(success, dark);
+        }
+
+        svg {
+            fill: palette(success, base);
+        }
+    }
+}
+
+.form-progress {
+    @include transition(all timing(xx-fast) ease-in-out 0s);
+}
+
+// ----------------------------
+// #INPUT TEXT
+// ----------------------------
+.input-text {
+    width: input-width(base);
+
+    &.input-alt {
+        padding: spacing-verical(small) 0;
+        border-width: 0 0 2px 0;
+
+        // STATE: is active or has focus
+        &:focus,
+        &.is-active {
+            border-color: $input-alt-border-focus;
+            background: $input-alt-background-focus;
+        }
+    }
+}
+
+// ----------------------------
+// #SELECT
+// ----------------------------
+.wrapper-replace-select {
+    position: relative;
+    // although this is on the wrapper, since we're positioning absolutley the replaced UI, we need to add a bottom margin here too
+    margin-bottom: spacing-vertical(small);
+}
+
+.input-select {
+    width: input-width(base);
+    height: rem($baseline*2);
+    line-height: spacing-vertical(small);
+
+    // set via JavaScript when a select menu has been replaced
+    // not to be used manually
+    &.is-replaced {
+        position: relative;
+        top: rem($baseline/4);
+        @include left(0);
+        z-index: z-index(mid-front);
+        opacity: 0;
+    }
+}
+
+// custom select styling
+// this is the replacement select menu
+.wrapper-custom-select {
+    @extend .field-input;
+    display: inline;
+    position: absolute;
+    top: 0;
+    @include left(0);
+    width: input-width(base);
+    line-height: 100%; // centers the text vertically while matching height of replacement menu
+    z-index: z-index(base);
+
+    // the color and position of the select menu icon
+    svg {
+        @include size(rem($baseline));
+        position: absolute;
+        @include right(rem(($baseline/4)));
+        margin-top: rem(-($baseline/10)); // vertically centers the down caret icon
+        fill: $input-control-color;
+    }
+
+    &:disabled,
+    &.is-disabled {
+
+        svg {
+            fill: $input-border-disabled;
+        }
+    }
+}
+
+// ----------------------------
+// #RADIO / CHECKBOX
+// ----------------------------
+.input-radio,
+.input-checkbox {
+    @include margin-right(spacing-horizontal(small));
+
+    // class used to use the enhanced version of checkboxes or radio buttons
+    &.replace-checkbox,
+    &.replace-radio {
+        @extend %a11y-hide;
+
+        & + .field-input-label:before {
+            @include size(rem($baseline/1.25));
+            display: inline-block;
+            @include margin-right(spacing-horizontal(x-small));
+            vertical-align: top;
+            content: "";
+        }
+    }
+
+    &.replace-checkbox {
+
+        & + .field-input-label:before {
+            @include input-checkbox;
```

Comment:
NIT: please add "()" whenever you are calling a mixin (to help denote that the item called is a mixin).